controllers/comments.controllers.js

```diff
@@ -46,3 +47,23 @@ exports.postComment = (req, res, next) => {
       }
     });
 };
+
+exports.deleteCommentById = (req, res, next) => {
+  const { comment_id } = req.params;
+
+  if (isNaN(Number(comment_id))) {
+    return res.status(400).send({ msg: 'Bad Request' });
+  }
+
+  removeCommentById(comment_id)
+    .then((deleteCount) => {
+      if (deleteCount === 0) {
+        return res.status(404).send({ msg: 'Comment Not Found' });
+      }
+      res.status(204).send();
+    })
+    .catch((err) => {
+      console.error('Error deleting comment:', err);
+      next(err);
+    });
+};
```

Comment:
Error handling should not be done in the controller. This should be done in the model then rejected so the .catch can pass everything on to your middle ware error handlers